src/plugins/i18n/server/i18n.js

```diff
@@ -0,0 +1,144 @@
+var fs = require('fs');
+var path = require('path');
+var process = require('child_process');
+var os = require('os');
+var kibanaPackage = require('../../../utils/package_json');
+
+const TRANSLATION_FILE_EXTENSION = 'json';
+const TRANSLATION_STORE_PATH = kibanaPackage.__dirname + '/data/store_translations';
+
+module.exports = {
+  storePluginLanguageTranslations: function (pluginName, pluginTranslationPath, language, cb) {
+    var translationFiles = [];
+    var languageList = [];
+    var translationStorePluginPath = module.exports.getPluginTranslationStoragePath(pluginName);
+    var translationFileName = language + '.' + TRANSLATION_FILE_EXTENSION;
+
+    module.exports.getPluginTranslationDetails(pluginTranslationPath, translationFiles, languageList, function (err) {
+      if (err) return cb (err);
+
+      var langSupported = false;
+      for (var langIndx in languageList) {
+        if (language === languageList[langIndx]) {
+          langSupported = true;
+          break;
+        }
+      }
+      if (!langSupported) {
+        return cb(Error(language + ' language is not supported in ' + pluginName + ' plugin.'));
+      }
+
+      try {
+        if (!fs.existsSync(translationStorePluginPath)) {
+          createDirectoriesRecursively(translationStorePluginPath);
+        }
+        for (var fileIndx in translationFiles) {
+          if (!translationFiles.hasOwnProperty(fileIndx)) continue;
+          var translationFile = translationFiles[fileIndx];
+          var pluginTranslationFileName = getFileName(translationFile);
+          if (pluginTranslationFileName !== translationFileName) continue;
+          var translationJson = require(translationFile);
+          var fileToWrite = translationStorePluginPath + '/' + translationFileName;
+          saveTranslationToFile(fileToWrite, translationJson);
+        }
+      } catch (err) {
+        return cb(err);
+      }
+    });
+
+    return cb(null);
+  },
+
+  getPluginLanguageTranslation: function (pluginName, language, callback) {
```

Comment:
seems like this could be synchronous - `fs.readFileSync` —  so that call flow of users doesn't have to change.

a later optimization could be some sort of cache here.